### PR TITLE
Deduplicate sub event array in super event on submission in back end

### LINKED ISSUE #834: [FEATURE] remove duplicated sub event on save the super event.
add a event as a subevent. before save it into a super event. user now can add more of this event. 
So the pr will remove the duplicated subevents on clicking the save button on event edit page.

### PR DESCRIPTION
**Pull request by clav1d**
## Description
Remove duplicates in the sub-event array in super events with back-end check.

**Fixes #834 **

[comment]: # (This project only accepts pull requests related to open issues.)
[comment]: # (If suggesting a new feature or change, please discuss it in an issue first.)
[comment]: # (If fixing a bug, there should be an issue describing it with steps to reproduce.)

## Review Time Estimate
_Please give your idea of how soon this pull request needs to be reviewed by selecting one of the options below. This can be based on the criticality of the issue at hand and/or other relevant factors._

[comment]: # (To select an option, please put an 'x' in the applicable box.)
[comment]: # (If you're unsure about any of these, don't hesitate to ask. We're here to help!.)

- [ ] Immediately
- [ ] Within a week
- [x] When possible

## Type of changes
_Please select a relevant option:_

[comment]: # (To select an option, please put an 'x' in the applicable box.)
[comment]: # (If you're unsure about any of these, don't hesitate to ask. We're here to help!.)

- [ ] Bug fix (non-breaking change which fixes an issue).
- [ ] New feature (non-breaking change which adds functionality).
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected).
- [ ] Other (any another change that does not fall in one of the above categories.)

## Checklist:
_Please select all applicable options:_

[comment]: # (To select your options, please put an 'x' in the all boxes that apply.)
[comment]: # (If you're unsure about any of these, don't hesitate to ask. We're here to help!.)

- [ ] I have signed the Rokwire Contributor License Agreement ([CLA](https://rokwire.org/rokwire_cla)). (_Any contributor who is not an employee of the University of Illinois whose official duties include contributing to the Rokwire software, or who is not paid by the Rokwire project, needs to sign the CLA before their contribution can be accepted._)
- [x] I have updated the [CHANGELOG](../CHANGELOG.md).
- [ ] I have read the [Contributor Guidelines](../CONTRIBUTING.md).
- [x] I have performed a self-review of my own code.
- [ ] I have commented my code, particularly in hard-to-understand areas.
- [ ] My change requires updating the documentation.
- [ ] I have made necessary changes to the documentation.
- [ ] I have added tests related to my changes.
- [ ] My changes generate no new warnings.
- [ ] New and existing unit tests pass locally with my changes.
- [ ] Any dependent changes have been merged and published in downstream modules.

---

[comment]: # (Template credit: This pull request template is based on Embedded Artistry {https://github.com/embeddedartistry/templates/blob/master/.github/PULL_REQUEST_TEMPLATE.md}, Clowder {https://github.com/clowder-framework/clowder/blob/develop/.github/PULL_REQUEST_TEMPLATE.md}, and TalAter {https://github.com/TalAter/open-source-templates} templates.)

### CODE REVIEW

**Comment by clav1d** [comment]
> @clav1d, as discussed earlier, please do not include PRs with linked issues in the GitHub Project. This creates duplicate entries. I'm updating this PR. Thanks.

I think I must have misclicked it. Sorry for the inconvenience.

File: CHANGELOG.md [hunk 1]
```diff
@@ -28,6 +28,7 @@ and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0
 - Add create event button on event detail page. [#818](https://github.com/rokwire/events-manager/issues/818)
 - Disable super event checkbox for sub events. [#833](https://github.com/rokwire/events-manager/issues/833)
 - Add confirmation modal when saving the published super events. [#819](https://github.com/rokwire/events-manager/issues/819)
+- Remove duplicates in the sub-event array in super events with back end check. [#834](https://github.com/rokwire/events-manager/issues/834)
```
Comment (sandeep-ps) [inline CHANGELOG.md:31]: Please move this entry to the Unreleased section above. Thanks.

Comment (sandeep-ps) [inline CHANGELOG.md:31]: Please move this entry to the Unreleased section above. Thanks.

**Comment by sandeep-ps** [comment]
I can see duplicate sub events in the super events page (pending super event). Please see the screenshot below:

image

I started with adding an event and marked `Is Super Event` to true and added another event as a sub-event twice. The above screenshot was taken after clicking the "Add" button.

File: CHANGELOG.md (hunk 1 above)
Comment (clav1d) [reply CHANGELOG.md]: I've moved this entry to the Unreleased section. Still working on fixing the bug.

**Comment by clav1d** [comment]
> I can see duplicate sub events in the super events page (pending super event). Please see the screenshot below:
> 
> image
> 
> I started with adding an event and marked `Is Super Event` to true and added another event as a sub-event twice. The above screenshot was taken after clicking the "Add" button.

Although I cannot reproduce this bug, I discovered and fixed another bug that will cause the application to raise exception during testing.

**Comment by sandeep-ps** [comment]
Basic tests look fine. Approving. Thanks.

**Review by sandeep-ps — APPROVED** [review]

**Comment by sandeep-ps** [comment]
Merging this PR. Thanks.

**Comment by sandeep-ps** [comment]
Merging this PR. Thanks.

**Pull request merged**